lxd/network/network_utils_sriov.go

```diff
@@ -70,17 +69,25 @@ func SRIOVGetHostDevicesInUse(s *state.State) (map[string]struct{}, error) {
 	err = s.Cluster.InstanceList(&filter, func(dbInst db.Instance, p db.Project, profiles []api.Profile) error {
 		// Expand configs so we take into account profile devices.
 		dbInst.Config = db.ExpandInstanceConfig(dbInst.Config, profiles)
-		dbInst.Devices = db.ExpandInstanceDevices(deviceConfig.NewDevices(dbInst.Devices), profiles).CloneNative()
+		for _, p := range profiles {
```

Comment:
@masnax why isnt this using `db.ExpandInstanceDevices` anymore?

I'm also confused why for the same set of inputs the new function can now return an error whereas before it could not.

Reply:
There's currently a lot of different types representing a set of Devices: 
* `deviceConfig.Device` which is a mask on `map[string]map[string]string`.
* `map[string]db.Device` which is the new database type for devices.
* `map[string]map[string]string` which is the old database type for devices.

I want to consolidate all these, but it would make this PR too big. For now,`db.ExpandInstanceDevices` returns a `deviceConfig.Device`, and modifying its arguments and return type would again increase the size of the PR. Just keeping the method in use everywhere would introduce a lot of redundant back and forth converting between the new and old types, so instead I'm temporarily manually expanding devices in some places. The goal is to eventually revert back to `db.ExpandInstanceDevices`, but with slightly different arguments and return types. 

As for the new error, previously `deviceConfig.NewDevices` was used to convert from the database representation of Devices: `map[string]map[string]string` to `deviceConfig.Devices`, but there was no actual validation to see if the received map is actually a valid device. Now I've introduced the `db.Device` struct which contains explicitly named fields with defined types like`db.DeviceType`, so validation can be performed.

Reply:
@masnax can we maybe have a FIXME or TODO so it's clear that it's a temporary step and more work is coming?

Reply:
@stgraber sure I'll add TODOs for those.